app/src/shared/components/Modal.tsx

```diff
@@ -0,0 +1,243 @@
+import React, { ReactNode, useEffect, useReducer, useRef } from 'react'
+import styled, { css, keyframes } from 'styled-components'
+import { useDiscardableEffect } from 'toasterhea'
+import gsap from 'gsap'
+import { SANS } from '../utils/styled'
+
+const bringIn = keyframes`
+    from {
+        opacity: 0;
+        transform: translateY(1%) translateZ(0) scale(0.98);
+    }
+    to {
+        opacity: 1;
+        transform: translateZ(0) scale(1) translateY(0);
+    }
+`
+
+const bringOut = keyframes`
+    from {
+        transform: translateZ(0) scale(1) translateY(0);
+    }
+    to {
+        transform: translateY(1%) translateZ(0) scale(0.98);
+    }
+`
+
+const fadeAway = keyframes`
+    from {
+        opacity: 1;
+    }
+    to {
+        opacity: 0;
+    }
+`
+
+export const RejectReason = {
+    CloseButton: Symbol('close button'),
+    Backdrop: Symbol('backdrop'),
+    EscapeKey: Symbol('escape key'),
+}
+
+export interface Props {
+    children?: ReactNode
+    title?: string
+    onReject?: (reason?: any) => void
+    onBeforeAbort?: (reason?: any) => boolean | null | void
+}
+
+export default function Modal({ title = 'Modal title', children, onReject, onBeforeAbort }: Props) {
+    const [dismissed, dismiss] = useReducer(() => true, false)
+
+    const rootRef = useRef<HTMLDivElement>(null)
+
+    useDiscardableEffect((discard) => {
```

Comment:
`onReject` settles the underlying promise but does not discard associated toastable component. You gotta tell it when it's ready to say bye bye.

Each toastable needs to call this hook to go away.

Pretty neat, if you think about it. Animations and cleanups are easy.